### PR TITLE
Merging two Linked List using c

### PR DESCRIPTION
**Pull request by saimanoharhm**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
yes
### Description

(Write your answer here.)
I have added code for Merging two Linked List using c
### Checklist

- [ yes] I've read the contribution guidelines.
- [ yes] I've checked the issue list before deciding what to submit.
- [ yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
#2171

### CODE REVIEW

**Comment by saimanoharhm** [comment]
@gat786 Could you pls review my PR

**Comment by gat786** [comment]
Add an example of what happens. 
Add a line break at the End Of File(EOF).
@saimanoharhm

**Comment by saimanoharhm** [comment]
@gat786 I didn't get you. What Example do you want me to add and break at eof

**Comment by saimanoharhm** [comment]
@gat786 Could you pls explain me more eloboratly

**Comment by vgb-codes** [comment]
@saimanoharhm Please update the [README.md](https://github.com/TesseractCoding/NeoAlgo/blob/master/C/README.md).

**Comment by HarshCasper** [comment]
Thanks for making your Pull Request!

PRs would be reviewed after the start of the Coding Period. Thanks!

**Comment by gat786** [comment]
@saimanoharhm An Example will be an example of how one would use your piece of code. Suppose I wrote a program to add two numbers then my example would be 

```
First Number: 12
Second Number: 25
Addition is: 37
```

Like a dummy example of how some one would run your piece of code.

And this [link](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline) should clear your mind on why we have a line break at the end of every file.

**Comment by saimanoharhm** [comment]
> @saimanoharhm Please update the [README.md](https://github.com/TesseractCoding/NeoAlgo/blob/master/C/README.md).

@gat786 I have already updated it. Pls check PR #2198

**Comment by saimanoharhm** [comment]
> @saimanoharhm An Example will be an example of how one would use your piece of code. Suppose I wrote a program to add two numbers then my example would be
> 
> ```
> First Number: 12
> Second Number: 25
> Addition is: 37
> ```
> 
> Like a dummy example of how some one would run your piece of code.
> 
> And this [link](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline) should clear your mind on why we have a line break at the end of every file.

@gat786 Ok I got it! but where do you want me to add it at the end of the program by putting output as a heading or where?

**Comment by saimanoharhm** [comment]
> @saimanoharhm Please update the [README.md](https://github.com/TesseractCoding/NeoAlgo/blob/master/C/README.md).

@vgb-codes  I have already updated it and I have made a PR for readme file updation also pls check PR #2198

**Comment by gat786** [comment]
Whereever you feel like it. Poeple usually add it at the bottom of their programs

**Comment by HarshCasper** [comment]
@saimanoharhm 

Please make your README updates here itself. Thanks!

**Comment by saimanoharhm** [comment]
@HarshCasper and @gat786 Could you check it now I have updated readme file and added example output as comment. Pls let me know if further changes needed

**Comment by saimanoharhm** [comment]
@HarshCasper Could you pls check my PR.

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Few changes: 

- Resolve the Merge Conflicts. 
- Add Comments above the lines of code wherever possible

**Comment by saimanoharhm** [comment]
@HarshCasper Could you pls check it now. I have done all the changes that you asked.

File: C/ds/Merging_two_Linked_List.c [hunk 1]
```diff
@@ -0,0 +1,127 @@
+/*
+
+This program is about merging two Linkedlist. Here user going to enter values of
+first and second linkedlist. It will merge the
+two linkedlists into one linkedlist by comparing the each and every node values
+and convert into one linkelist in the ascending order.
```
Comment (HarshCasper) [inline C/ds/Merging_two_Linked_List.c]: Not gramatically correct: 
- Replace `Linkedlist` with `Linked List`
- Here `the` user `is` going...

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Use a 4-spaces formatting

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Few more changes: 
- Don't use variable names like `A`, `B`, `t` and all. Use meaningful variable names
- Share the complexity of the code
- Put spaces after functions 
- Remove some redundant comments

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**